cts/schemas/test-3/ref.err/acl-drop.ref.err-99

```diff
@@ -0,0 +1 @@
+WARNING: CIB syntax changes may invalidate ACLs that use "xpath". It is strongly recommended to run "cibadmin --upgrade" and then examine the updated CIB carefully to ensure ACLs still match the desired intent.
```

Comment:
I'm a little concerned about unnecessarily alarming users who aren't using any deprecated syntax. Is there a simple way to check whether we made any changes?

Reply:
I can't think of a *simple* way to do that within the XSLT stylesheets, especially since we're running multiple stylesheets -- this final stylesheet doesn't know what was done in earlier stylesheets.

It might be possible to add a new `<change>` element to a place where something has changed, and then drop all the `<change>` elements at the end. It's clunky but manageable.

Setting a boolean "changed" attribute at the top of the document would be cleaner in theory. But due to the way XSLT works, it's even uglier. A template that matches a child can't change the behavior of its parent. So there would have to be a top-level template that duplicates all the logic in the other templates, to decide whether to set the "changed" flag.

The C code could perhaps compare the original XML to the transformed XML, minus the validate-with attribute, etc.  That may turn out to be more complex, IDK, and the warning wouldn't appear after manually running the transformation.

Reply:
A combination of `<change>` elements (to replace dropped elements) and `"changed"` attributes (as siblings of changed attributes) might actually work. It definitely adds clutter but it might be worth it; doesn't seem complicated to implement.

I'll add another commit soon as a demo and see if we want that or not.

Reply:
Use the transform prefix in case we add a real changed name someday

Reply:
Shouldn't matter, since it's not allowed by the old schema that will be the input to the transformation. Similar to how we're using "original" attribute within the pipeline to deal with id-refs